garden/src/features/modal/components/edit/EditModalFunctionFormFields.tsx

```diff
@@ -176,6 +183,81 @@ return my_garden.function_name(input)`}`;
           />
         </div>
       </section>
+
+      <section>
+        <h2 className="mb-6 border-b pb-2 text-xl font-bold text-gray-800">Associated Materials</h2>
+        <div className="space-y-8">
+          <AssociatedMaterialsSection fieldName="papers" resourceType="paper" />
```

Comment:
The AssociatedMaterialsSection component feels odd and un-reacty. I don't think there should be a big switch statement inside a component. I think you can do something a lot more functional/clean here